lib/data-types.js

```diff
@@ -481,6 +534,33 @@ DATEONLY.prototype._stringify = function _stringify(date) {
 };
 
 /**
+ * A YEAR column.
+ *
+ *
+ * @property YEAR
+ */
+function YEAR() {
+  if (!(this instanceof YEAR)) return new YEAR();
+}
+util.inherits(YEAR, ABSTRACT);
+
+YEAR.prototype.key = YEAR.key = 'YEAR';
+YEAR.prototype.toSql = function() {
+  return 'YEAR';
+};
+
+YEAR.prototype._stringify = function _stringify(date) {
+  return date;
```

Comment:
I think we need to return year in string format like `2016` not date(if its not an string already), please add some test for this like https://github.com/anonmily/sequelize/blob/4d32f18526b49e5b586fd9ddc037588d94803918/test/integration/data-types.test.js#L441 to verify that we indeed return year in string format